config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/PrivateKey.java

```diff
@@ -136,10 +136,13 @@ public void postLoad(String entityKey, Bundle bundle, File rootFolder, IdGenerat
         setKeyStoreType(KeyStoreType.fromName(getKeystore()));
         if (bundle.getPrivateKeyFiles().get(getAlias()) != null) {
             setPrivateKeyFile(bundle.getPrivateKeyFiles().get(getAlias()));
-        }
-        if (rootFolder != null && getPrivateKeyFile() == null) {
+        } else if (rootFolder != null && getPrivateKeyFile() == null) {
             loadPrivateKey(this, new File(rootFolder, "config/privateKeys"), false);
         }
+
+        if (getPrivateKeyFile() != null) {
+            bundle.getPrivateKeyFiles().put(getAlias(), getPrivateKeyFile());
```

Comment:
I think it should be done inside the above else-if block.

Reply:
@graju256 Fixed!